pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,313 @@
+
+	req, err := http.NewRequest("GET", statsURL, nil)
+	if err != nil {
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	res, err := s.client.Do(req)
+	if res == nil || err != nil {
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case 200:
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			pulsarLog.Error(err, "error requesting stats from url")
+			return nil, fmt.Errorf("error requesting stats from url: %s", err)
+		}
+		err = json.Unmarshal(body, stats)
+		if err != nil {
+			pulsarLog.Error(err, "error unmarshalling response")
+			return nil, fmt.Errorf("error unmarshalling response: %s", err)
+		}
+		return stats, nil
+	case 404:
+		return nil, nil
+	default:
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+}
+
+func (s *pulsarScaler) getMsgBackLog() (int, bool, error) {
+	stats, err := s.GetStats()
+	if err != nil {
+		return 0, false, err
+	}
+
+	if stats == nil {
+		return 0, false, nil
+	}
+
+	v, found := stats.Subscriptions[s.metadata.subscription]
+
+	return v.Msgbacklog, found, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *pulsarScaler) IsActive(_ context.Context) (bool, error) {
+	msgBackLog, found, err := s.getMsgBackLog()
+	if err != nil {
+		return false, err
+	}
+
+	if !found || msgBackLog == 0 {
+		pulsarLog.Info("Pulsar subscription is not active, either no subscription found or no backlog detected")
+		return false, nil
+	}
+
+	pulsarLog.Info("Pulsar subscription is active with backlog")
+	return true, nil
+}
+
+// GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *pulsarScaler) GetMetrics(_ context.Context, metricName string, _ labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	msgBacklog, found, err := s.getMsgBackLog()
+	if err != nil {
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	if !found {
+		return nil, nil
+	}
+
+	pulsarLog.Info(fmt.Sprintf("Pulsar MsgBacklog: %d", msgBacklog))
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(msgBacklog), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *pulsarScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetMetricValue := resource.NewQuantity(s.metadata.msgBacklogThreshold, resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s-%s", "pulsar", s.metadata.topic, s.metadata.subscription)),
```

Comment:
For names, [we use the internal function](https://github.com/kedacore/keda/blob/main/pkg/scalers/scaler.go#L110) that ensures the metric name is unique inside the scaler, you can find an example [here](https://github.com/kedacore/keda/blob/main/pkg/scalers/rabbitmq_scaler.go#L466). The whole process is defined in the docs about [GetMetricsSpecForScaling](https://github.com/kedacore/keda/blob/main/CREATE-NEW-SCALER.md#getmetricspecforscaling)